L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -0,0 +1,938 @@
+    layer -= heOffset_;
+    subdet = ForwardSubdetector::HGCHEF;
+  }
+  packed_value |= ((layer & HGCalDetId::kHGCalLayerMask) << HGCalDetId::kHGCalLayerOffset);
+  packed_value |= ((module & HGCalDetId::kHGCalWaferMask) << HGCalDetId::kHGCalWaferOffset);
+  packed_value |= ((subdet & DetId::kSubdetMask) << DetId::kSubdetOffset);
+  return packed_value;
+}
+
+void HGCalTriggerGeometryV9Imp3::unpackWaferId(unsigned wafer, int& waferU, int& waferV) const {
+  unsigned waferUAbs = (wafer >> HGCSiliconDetId::kHGCalWaferUOffset) & HGCSiliconDetId::kHGCalWaferUMask;
+  unsigned waferVAbs = (wafer >> HGCSiliconDetId::kHGCalWaferVOffset) & HGCSiliconDetId::kHGCalWaferVMask;
+  waferU = (((wafer >> HGCSiliconDetId::kHGCalWaferUSignOffset) & HGCSiliconDetId::kHGCalWaferUSignMask) ? -waferUAbs
+                                                                                                         : waferUAbs);
+  waferV = (((wafer >> HGCSiliconDetId::kHGCalWaferVSignOffset) & HGCSiliconDetId::kHGCalWaferVSignMask) ? -waferVAbs
+                                                                                                         : waferVAbs);
+}
+
+void HGCalTriggerGeometryV9Imp3::unpackLayerWaferId(unsigned wafer, unsigned& layer, int& waferU, int& waferV) const {
+  layer = (wafer >> HGCalModuleDetId::kHGCalLayerOffset) & HGCalModuleDetId::kHGCalLayerMask;
+  waferU = (wafer >> HGCalModuleDetId::kHGCalModuleUOffset) & HGCalModuleDetId::kHGCalModuleUMask;
+  waferV = (wafer >> HGCalModuleDetId::kHGCalModuleUOffset) & HGCalModuleDetId::kHGCalModuleUMask;
+}
+
+void HGCalTriggerGeometryV9Imp3::etaphiMappingFromSector0(int& ieta, int& iphi, unsigned sector) const {
+  if (sector == 0) {
+    return;
+  }
+
+  iphi = iphi + (sector * hSc_num_panels_per_sector_);
+}
+
+HGCalGeomRotation::WaferCentring HGCalTriggerGeometryV9Imp3::getWaferCentring(unsigned layer) const {
+  if (layer <= 28) {  // CE-E
+    return HGCalGeomRotation::WaferCentring::WaferCentred;
+  } else if ((layer % 2) == 1) {  // CE-H Odd
+    return HGCalGeomRotation::WaferCentring::CornerCentredY;
+  } else {  // CE-H Even
+    return HGCalGeomRotation::WaferCentring::CornerCentredMercedes;
+  }
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::etaphiMappingToSector0(int& ieta, int& iphi) const {
+  unsigned sector = 0;
+
+  if (unsigned(std::abs(iphi)) >= 2 * hSc_num_panels_per_sector_)
+    sector = 2;
+  else if (unsigned(std::abs(iphi)) >= hSc_num_panels_per_sector_)
+    sector = 1;
+  else
+    sector = 0;
+
+  iphi = iphi - (sector * hSc_num_panels_per_sector_);
+
+  return sector;
+}
+
+bool HGCalTriggerGeometryV9Imp3::validTriggerCell(const unsigned trigger_cell_id) const {
+  return validTriggerCellFromCells(trigger_cell_id);
+}
+
+bool HGCalTriggerGeometryV9Imp3::disconnectedModule(const unsigned module_id) const {
+  bool disconnected = false;
+  if (disconnected_modules_.find(HGCalDetId(module_id).wafer()) != disconnected_modules_.end())
+    disconnected = true;
```

Comment:
Should be removed for the moment as it is not consistent with the new module id

Reply:
Removed